GillianCore/GIL_Syntax/Annot.ml

```diff
@@ -1,41 +1,17 @@
-(** {b GIL annot}. *)
-type expansion_kind = NoExpansion | Function of string [@@deriving yojson]
-
-type t = {
-  origin_loc : Location.t option;  (** Better not to know what this is for *)
-  origin_id : int option;  (** Origin Id, that should be abstracted away *)
-  loop_info : string list; [@default []]
-  hidden : bool; [@default false]
-      (** Should this command be hidden when debugging? *)
-  expansion_kind : expansion_kind; [@default NoExpansion]
-      (** Should this command be expanded when lifting? (i.e. loops to functions in WISL) *)
-  (* TODO: move the following fields to WISL-specific annot *)
-  loop_prefix : bool; [@default false]
-  end_of_cmd : bool; [@default false]
-  is_return : bool; [@default false]
-}
-[@@deriving yojson, make]
-
-let get_loop_info (annot : t) = annot.loop_info
-
-let set_loop_info (annot : t) (loop_info : string list) =
-  { annot with loop_info }
-
-let get_origin_loc annot = annot.origin_loc
-let get_origin_id annot = annot.origin_id
-let is_hidden (annot : t) = annot.hidden
-let hide (annot : t) = { annot with hidden = true }
-let get_expansion_kind (annot : t) = annot.expansion_kind
-
-let set_expansion_kind expansion_kind (annot : t) =
-  { annot with expansion_kind }
-
-let is_loop_prefix (annot : t) = annot.loop_prefix
-
-let set_loop_prefix ?(is_prefix = true) (annot : t) =
-  { annot with loop_prefix = is_prefix }
-
-let is_end_of_cmd (annot : t) = annot.end_of_cmd
-let set_end_of_cmd (annot : t) = { annot with end_of_cmd = true }
-let is_return (annot : t) = annot.is_return
-let set_return (annot : t) = { annot with is_return = true }
+module type S = sig
+  type t [@@deriving yojson]
+
+  val from_loc : ?origin_loc:Location.t -> unit -> t
+  val get_origin_loc : t -> Location.t option
+  val get_loop_info : t -> string list
+  val is_hidden : t -> bool
+end
+
+module Default = struct
+  type t = Location.t option [@@deriving yojson]
+
+  let from_loc ?origin_loc () : t = origin_loc
+  let get_origin_loc t = t
+  let get_loop_info (_ : t) = []
+  let is_hidden (_ : t) = false
+end
```

Comment:
The default should really contain a `loop_info`, so that users can just use it directly for their own TL until they need more

Reply:
i.e.
```ocaml
type t = {
  loop_info: string list;
  location: Location.t option
}
```

Reply:
Ah wrote, this comment and the `CAnnot.ml` comment 2 days apart, didn't realise
So yeah, `Default` should be what's current `CAnnot` and C should just use the Default one

Reply:
a582d1000901abde15e95196861bac438b0bd6da